R/ResamplingForecastHoldout.R

```diff
@@ -0,0 +1,82 @@
+#' @title Holdout Resampling
+#'
+#' @usage NULL
+#' @name mlr_forecast_resamplings_holdout
+#' @format [R6::R6Class] inheriting from [Resampling].
+#' @include ResamplingForecastHoldout.R
+#'
+#' @section Construction:
+#' ```
+#' ResamplingForecastHoldout$new()
+#' mlr_resamplings$get("forecast.holdout")
+#' rsmp("forecast.holdout")
+#' ```
+#'
+#' @description
+#' Splits data into a training set and a test set.
+#' Parameter `ratio` determines the ratio of observation going into the training set (default: 2/3).
+#'
+#' @section Fields:
+#' See [Resampling].
+#'
+#' @section Methods:
+#' See [Resampling].
+#'
+#' @section Parameters:
+#' * `ratio` :: `numeric(1)`\cr
+#'   Ratio of observations to put into the training set.
+#'
+#' @references
+#' \cite{mlr3}{bischl_2012}
+#'
+#' @template seealso_resampling
+#' @export
+#' @examples
+#'  #Create a task with 10 observations
+#' task = mlr3::tsk("airpassengers")
+#' task$filter(1:10)
+#'
+#' #Instantiate Resampling
+#' rfho = mlr3::rsmp("forecast.holdout", ratio = 0.5)
+#' rfho$instantiate(task)
+#'
+#' # Individual sets:
+#' rfho$train_set(1)
+#' rfho$test_set(1)
+#' intersect(rfho$train_set(1), rfho$test_set(1))
+#'
+#' # Internal storage:
+#' rfho$instance # simple list
+ResamplingForecastHoldout = R6Class("ResamplingForecastHoldout", inherit = Resampling,
+  public = list(
+    initialize = function() {
+      ps = ParamSet$new(list(
+        ParamDbl$new("ratio", lower = 0, upper = 1, tags = "required")
+      ))
+      ps$values = list(ratio = 2 / 3)
+
+      super$initialize(id = "forecast.holdout", param_set = ps, man = "mlr3::mlr_resamplings_forecast.holdout")
+    },
+
+    iters = 1L
+  ),
+
+  private = list(
+    .sample = function(ids) {
+      nr = round(length(ids) * self$param_set$values$ratio)
+      ii = ids[1:nr]
```

Comment:
Here you want to check that no time IDs exist in both the training and test set so there is no leakage. If you had APPL and MSFT stock prices and accidentally put AAPL at 4:00 PM into your training and MSFT at 4:00 PM into your testing there would be time leakage across the test/training sets

Reply:
I don't think I understand the issue. In this example 4:00 PM would be e.g. id = 10 for both AAPL and MSFT. The train set is then 1:10 and the test set would start at id=11.

Reply:
If the data is wide yes, but if the data is long wouldn't the IDs be 10 and 11. See the comment [here](https://github.com/mlr-org/mlr3forecasting/pull/17#issuecomment-577402058) for an example. I only glanced at the code so if your sure there's no leakage here you can mark it resolved.